apps/bsatool/CMakeLists.txt

```diff
@@ -18,3 +18,7 @@ if (BUILD_WITH_CODE_COVERAGE)
   add_definitions (--coverage)
   target_link_libraries(bsatool gcov)
 endif()
+
+if(MSVC)
+  target_link_libraries(bsatool imm32.lib winmm.lib version.lib)
```

Comment:
Okay, I tracked down the problem.  Changing `${SDL_LIBRARY}` to `${SDL_LIBRARIES}` in the CMakeList for the `components` target fixes everything.  I have done this and removed all of my dirty work-arounds in the latest commit.